### PR TITLE
Issue 1412: Trigger a modal on domain request submission

### LINKED ISSUE #1412: Update "Submit your domain request" button behavior
### Issue description

As a Registrant Applicant, I want to understand the importance and implications upon clicking the final submit button so that I'm made aware that this is the final stage and I will no longer be allowed to edit my application.

Implement solution for "Submit your domain request" button base on current design.
Figma showing showing user experience: [Prototype](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2314%3A49899&type=design&node-id=2314-50557&viewport=-2156%2C155%2C0.32&t=7Mt0wTODqUSsxU1o-1&scaling=min-zoom&starting-point-node-id=2314%3A50557&mode=design) and [Screens](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?type=design&node-id=2314-49899&mode=design)

### Acceptance criteria

- [ ] Clicking "Submit your domain request" button triggers a modal
- [ ] The modal headline says: `You are about to submit a domain request for <requesteddomain.gov>. `
- [ ] The body says: `Once you submit this request, you won’t be able to make further edits until it’s reviewed by our staff. You’ll only be able to withdraw your request. `
- [ ] The button says: `Submit request`
- [ ] The link button says: `Go back`
- [ ] User cannot click out of modal, they can exit back to the Review page by clicking "Go back" or the "X" icon. 
- [ ] Clicking the primary button submits their request and shows them the Confirmation page

### Additional context

_No response_

### Links to other issues

🔄 Corresponding design ticket #767

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1412 

## Changes

- Implement a forced action modal between domain request submit and form submit button,
- Unit test that looks for the modal trigger and modal data,
-  Fixes for existing DA unit tests that query and trigger the previously unique form.

## Context for reviewers

This ticket made me wonder whether it's worth adding Selenium to our tech stack.

## Setup

If you want to test locally (or just have a better experience going through a pre-populates domain request), click on a request then change the last part of the URL to review: http://localhost:8080/register/review/

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/.pa11yci [hunk 1]
```diff
@@ -19,7 +19,6 @@
         "http://localhost:8080/register/other_contacts/",
         "http://localhost:8080/register/anything_else/",
         "http://localhost:8080/register/requirements/",
-        "http://localhost:8080/register/review/",
```
Comment (rachidatecs) [inline src/.pa11yci:22]: pa11y does not like a form with no submit. ANDI does not complain.

File: src/registrar/tests/test_copy_names_from_contacts_to_users.py [hunk 2]
```diff
@@ -18,8 +18,8 @@ def setUp(self):
         self.user2 = User.objects.create(username="user2")
         self.user3 = User.objects.create(username="user3")
         self.user4 = User.objects.create(username="user4")
-        # The last user created triggers the creation of a contact and attaches itself to it. @Neil wth is going on?
-        # This bs_user defuses that situation so we can test the code.
+        # The last user created triggers the creation of a contact and attaches itself to it. See signals.
```
Comment (neilmb) [inline src/registrar/tests/test_copy_names_from_contacts_to_users.py:21]: **question (non-blocking)**: I don't see how this is relevant to this PR? Did this come over from another branch?

**Review by neilmb — APPROVED** [review]
LGTM. Sorry you had to change so many things in the unit tests that assumed a single form on the page.

File: src/registrar/tests/test_copy_names_from_contacts_to_users.py (hunk 2 above)
Comment (rachidatecs) [reply src/registrar/tests/test_copy_names_from_contacts_to_users.py:21]: I remembered that I'd left the @Neil comment in the previous PR and thought it's ok to sneak in a comment clean up

**Review by gabydisarli — APPROVED** [review]

**Pull request merged**